src/dlgProfilePreferences.cpp

```diff
@@ -2309,6 +2372,205 @@ void dlgProfilePreferences::slot_handleHostDeletion(Host* pHost)
+        }
+        QTableWidgetItem* pRoomNumbers = new QTableWidgetItem(roomNumberStringList.join(QStringLiteral(", ")));
+
+        QToolButton * pDummyButton = new QToolButton();
+        if (isSingleFontUsable) {
+            pSymbolInFont->setText(symbol);
+            pSymbolAnyFont->setText(symbol);
+            pDummyButton->setIcon(QIcon(QStringLiteral(":/icons/dialog-ok-apply.png")));
+        } else {
+            // Need to switch to a different font as it is possible that the
+            // single font may not have the replacement glyph either...!
+            pSymbolInFont->setFont(anyFont);
+            pSymbolInFont->setText(QString(QChar::ReplacementCharacter));
+            if (isAllFontUsable) {
+                pSymbolAnyFont->setText(symbol);
+                pDummyButton->setIcon(QIcon(QStringLiteral(":/icons/dialog-warning.png")));
+            } else {
+                pSymbolAnyFont->setText(QString(QChar::ReplacementCharacter));
+                pDummyButton->setIcon(QIcon(QStringLiteral(":/icons/dialog-error.png")));
+            }
+        }
+        pTableWidget->setCellWidget(++row, 0, pDummyButton);
+
+        pTableWidget->setItem(row, 1, pSymbolInFont);
+        pTableWidget->setItem(row, 2, pSymbolAnyFont);
+        pTableWidget->setItem(row, 3, pCodePointDisplay);
+        pTableWidget->setItem(row, 4, pUsageCount);
+        pTableWidget->setItem(row, 5, pRoomNumbers);
+    }
+    pTableWidget->sortItems(4, Qt::DescendingOrder);
+    pTableWidget->setSortingEnabled(true);
+    pTableWidget->resizeColumnsToContents();
+    // The room number column can contain a lot of rooms...
+    pTableWidget->resizeRowsToContents();
+    mpDialogMapGlyphUsage->show();
+    mpDialogMapGlyphUsage->raise();
+}
+
+void dlgProfilePreferences::slot_showMapGlyphUsage()
+{
+    if (!mpHost || !mpHost->mpMap) {
+        return;
+    }
+
+    if (mpDialogMapGlyphUsage) {
+        // Already in use so just refresh the contents instead:
+        generateMapGlyphDisplay();
+        return;
+    }
+
+    QUiLoader loader;
+    QFile file(QStringLiteral(":/ui/glyph_usage.ui"));
+    file.open(QFile::ReadOnly);
+    mpDialogMapGlyphUsage = qobject_cast<QDialog*>(loader.load(&file, this));
+    file.close();
+    if (!mpDialogMapGlyphUsage) {
+        qWarning() << "dlgProfilePreferences::slot_showMapGlyphUsage() ERROR: failed to create the dialog!";
+        return;
+    }
+
+    mpDialogMapGlyphUsage->setWindowIcon(QIcon(QStringLiteral(":/icons/place_of_interest.png")));
+    mpDialogMapGlyphUsage->setWindowTitle(tr("Map symbol usage - %1").arg(mpHost->getName()));
+    mpDialogMapGlyphUsage->setAttribute(Qt::WA_DeleteOnClose);
+    generateMapGlyphDisplay();
+}
+
+void dlgProfilePreferences::slot_setMapSymbolFontStrategey(const bool isToOnlyUseSelectedFont)
+{
+    Host* pHost = mpHost;
+    if (!pHost ||!pHost->mpMap) {
+        return;
+    }
+
+    if (pHost->mpMap->mIsOnlyMapSymbolFontToBeUsed != isToOnlyUseSelectedFont) {
```

Comment:
"if is only map symbol font to be used it not is to only use selected font"... this hurts my brain to read. Could the variable names be more programmer-friendly to improve readability?

Reply:
This `if` is needed to run the enclose chunk only if the setting has changed - the two variables do represent the same setting but when I did the dialog/form I had not yet created the C++ `TMap` variable... Those variable names are programmer friendly IMVHO c.v. `checkBox_USE_IRE_DRIVER_BUGFIX` - AT LEAST I AM NOT SHOUTING. 😉

Reply:
I didn't create that variable name and we don't use that style anymore for a long while - it's not really a valid reason...

how about "if only use specified map font"? It sounds a lot better than "if is only map symbol font to be used"...